app/controllers/foreman_gutterball/api/v2/content_reports_controller.rb

```diff
@@ -4,29 +4,46 @@ module V2
       class ContentReportsController < ::Katello::Api::V2::ApiController
         api :GET, '/content_reports', 'List available reports'
         def index
-          service.available_reports
+          @report = service.report_details
+          render :json => @report
         end
 
-        api :GET, '/content_reports/consumer_status', 'Consumer status reports'
+        api :GET, '/content_reports/:id', 'Report details'
+        def show
+          @report = service.report_details params[:id]
+          render :json => @report
+        end
+
+        api :GET, '/content_reports/:id/run', 'Generate reports'
+        def run
+          query_params = request.query_parameters.symbolize_keys
+          validate_params(query_params.keys) unless query_params.empty?
+          @report = service.run_reports(params[:id], query_params)
+          render :json => @report
+        end
+
+        private
+
         def consumer_status
-          service.consumer_status
+          [:consumer_uuid, :owner, :status, :on_date, :page, :per_page, :custom]
         end
 
-        api :GET, '/content_reports/consumer_trend', 'Consumer trend reports'
         def consumer_trend
-          service.consumer_trend
+          [:consumer_uuid, :hours, :start_date, :end_date, :custom]
         end
 
-        api :GET, '/content_reports/status_trend', 'Status trend reports'
         def status_trend
-          service.status_trend
+          [:start_date, :end_date, :owner, :sku, :subscription_name, :management_enabled, :timezone]
         end
 
-        private
-
         def service
           GutterballService.new
         end
+
+        def validate_params(query_keys)
+          report_keys = send(params[:id])
+          fail Katello::Errors::NotFound, _('Invalid parameters') unless report_keys & query_keys == query_keys
```

Comment:
This error should probably be `HttpErrors::BadRequest` since the client/user is sending invalid parameters to the server.